main.tf

```diff
@@ -8,10 +8,13 @@ locals {
   waf_v2_policies                        = local.enabled && length(var.waf_v2_policies) > 0 ? { for policy in flatten(var.waf_v2_policies) : policy.name => policy } : {}
   dns_firewall_policies                  = local.enabled && length(var.dns_firewall_policies) > 0 ? { for policy in flatten(var.dns_firewall_policies) : policy.name => policy } : {}
   network_firewall_policies              = local.enabled && length(var.network_firewall_policies) > 0 ? { for policy in flatten(var.network_firewall_policies) : policy.name => policy } : {}
+
+
+  logging_configuration                  = local.enabled && var.firehose_enabled ? { logDestinationConfigs:[join("", aws_kinesis_firehose_delivery_stream.firehose_stream.*.id)]} : local.enabled && var.firehose_arn != null ? { logDestinationConfigs:[var.firehose_arn], redactedFields:[{ redactedFieldType:"SingleHeader", redactedFieldValue:"Cookies"},{redactedFieldType:"Method"}]} : {}
```

Comment:
difficult to read and understand.
maybe split into multiple lines (and split into a few locals)